clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepository.java

```diff
@@ -172,6 +181,7 @@ public Store refreshOneStore(String storeName) {
       if (newStore != null && !storeConfig.isDeleting()) {
```

Comment:
why do we aggresively remove the store in the `else` part here? A metadata fetch call fail would not return null also.

Reply:
Its not related to this PR, seems like we reconstruct a branch new store on every refresh,  even though we just to look at the store properties only.

Reply:
Right, the store deletion path is very fragile (another reason to move away from meta system store). In reality it's very possible that the system store is already deleted together with the user store before the NativeMetadataRepository even got a chance to refresh. Meaning the TC will throw exception even earlier on `getAndSetStoreConfigFromSystemStore`. Right now the only case this `else` block can be executed is when the `newStore` object is not null and when `storeConfig.isDeleting()` is true. This is because we will throw exception if the fetch is actually null in `getStoreMetaValue`:
```
    if (value == null) {
      throw new MissingKeyInStoreMetadataException(key.toString(), StoreMetaValue.class.getSimpleName());
    }
```
This ensures we won't accidentally remove store when encountering a null value from TC.